pallets/identity/src/claims.rs

```diff
@@ -481,6 +481,16 @@ impl<T: Config> Module<T> {
         // Sender has to be part of CDDProviders
         Self::ensure_authorized_cdd_provider(cdd_did)?;
 
+        let record = <DidRecords<T>>::get(cdd_did);
+        let cost = secondary_keys.iter().fold(0usize, |cost, auth| {
+            //Check permissions limits
+            Self::ensure_perms_length_limited(&auth.permissions);
+            cost.saturating_add(auth.permissions.complexity())
+        });
```

Comment:
The `Result` from line `Self::ensure_perms_length_limited(&auth.permissions);` is ignored, so that check doesn't do anything.

You can use `iter().try_fold` instead to allow returning a `Result`.

Reply:
alright would make the change.

Reply:
I got an error saying "the trait bound `usize: Try` is not satisfied" using `iter().try_fold` so I called `unwrap()` on the statement itself.

Reply:
`unwrap()` only stops the compiler warning, but doesn't handle the error.

I wrote an example of how to use `try_fold` below.